hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/forensics/OrderedComparison.java

```diff
@@ -68,40 +68,68 @@ public static List<DifferingEntries> findDifferencesBetweenV6(
         System.out.println(" ➡️  Read " + firstEntries.size() + " entries");
         System.out.println("Parsing stream @ " + secondStreamDir + "(including " + inclusionDescription + ")");
         final var secondEntries = parseV6RecordStreamEntriesIn(secondStreamDir, inclusionTest);
-        List<RecordStreamEntry> newSecondEntries = getNewSecondRecordStreamEntries(firstEntries, secondEntries);
         System.out.println(" ➡️  Read " + secondEntries.size() + " entries");
-        int missed = newSecondEntries.size() - secondEntries.size();
-        if (missed > 0) {
-            System.out.println(" ➡️  Missed " + missed + " entries");
-        } else if (missed < 0) {
-            System.out.println(" ➡️  Added " + (-missed) + " entries");
-            List<RecordStreamEntry> newFirstEntries = getNewSecondRecordStreamEntries(secondEntries, firstEntries);
-            return diff(newFirstEntries, secondEntries, recordDiffSummarizer);
-        }
-        return diff(firstEntries, newSecondEntries, recordDiffSummarizer);
+        final var compareList = getCompareList(firstEntries, secondEntries);
+        return diff(compareList.firstList, compareList.secondList, recordDiffSummarizer);
     }
 
+    record CompareList(@NonNull List<RecordStreamEntry> firstList, @NonNull List<RecordStreamEntry> secondList) {}
+    ;
+
     @NonNull
-    private static List<RecordStreamEntry> getNewSecondRecordStreamEntries(
+    private static CompareList getCompareList(
             List<RecordStreamEntry> firstEntries, List<RecordStreamEntry> secondEntries) {
-        List<RecordStreamEntry> ret = new ArrayList<>();
-        if (secondEntries.isEmpty()) {
-            return ret;
-        }
-        RecordStreamEntry firstEntry;
-        RecordStreamEntry secondEntry;
-        int secondIndex = 0;
-        for (RecordStreamEntry entry : firstEntries) {
-            firstEntry = entry;
-            secondEntry = secondEntries.get(secondIndex);
-            if (secondEntry.consensusTime().equals(firstEntry.consensusTime())) {
-                ret.add(secondEntry);
-                secondIndex++;
-            } else {
-                ret.add(new RecordStreamEntry(null, null, firstEntry.consensusTime()));
+        List<RecordStreamEntry> firstList = new ArrayList<>();
+        List<RecordStreamEntry> secondList = new ArrayList<>();
+
+        if (secondEntries.isEmpty() || firstEntries.isEmpty()) {
+            return new CompareList(firstEntries, secondEntries);
+        } else {
+            int i = 0, j = 0;
```

Comment:
Nit: codacy is going to start flagging these soon, based on the PRs I've seen from Nathan Klick
```
int i = 0;
int j = 0;
```

Reply:
I don't completely agree with codacy. Fixed it here.